core/src/main/java/org/mule/runtime/core/internal/util/message/MessageUtils.java

```diff
@@ -214,15 +214,12 @@ public static Message toMessage(Result<?, ?> result,
   private static Message toMessage(Result<?, ?> result, DataType dataType, Object value) {
     Message.Builder builder = Message.builder().payload(new TypedValue<>(value, dataType, result.getByteLength()));
 
-    if (result.getAttributes().isPresent()) {
-      Object att = result.getAttributes().get();
-
+    result.getAttributes().ifPresent(att -> {
```

Comment:
this was on purpose because of performance, to avoid instantiating the lambda for each message built